Session/Settings/AppIconViewModel.swift

```diff
@@ -189,4 +195,19 @@ class AppIconViewModel: SessionTableViewModel, NavigatableStateHolder, Observabl
         
         selectedOptionsSubject.send(icon?.rawValue)
     }
+    
+    private func restorePreviousIcon(_ identifier: String??) {
+        var previousIcon: AppIcon? {
+            if let previousIcon = identifier {
+                // Set previous app icon
+                return AppIcon(name: previousIcon)
+            } else if let previousIcon = dependencies[defaults: .standard, key: .lastSelectedAppIconDisguise] {
+                // Handles app close instance to restore previously selected
+                return AppIcon(name: previousIcon)
+            }
+            return .weather
+        }
+        
+        updateAppIcon(previousIcon)
```

Comment:
Since we are setting `lastSelectedAppIconDisguise` every time the icon is changed handling `previous` seems redundant - can you instead please update the `.appIconUseAlternate` `onTap` to just be:
```Swift
onTap: { [weak self] in
    let lastSelected: String? = dependencies[defaults: .standard, key: .lastSelectedAppIconDisguise]
    
    switch current {
        case .some: self?.updateAppIcon(nil)
        case .none:
            self?.updateAppIcon(lastSelected.map { AppIcon(name: $0) } ?? .weather)
    }
}
```